### PR TITLE
glrnvim: init at 0.1.1

### PR DESCRIPTION
**Pull request by sgraf812**
###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/applications/editors/neovim/glrnvim/default.nix [hunk 1]
```diff
@@ -0,0 +1,28 @@
+{ stdenv, rustPlatform, makeWrapper, alacritty, neovim }:
+
+rustPlatform.buildRustPackage rec {
+  pname = "glrnvim";
+  version = "0.1.1";
+
+  src = builtins.fetchTarball {
```
Comment (JohnAZoidberg) [inline pkgs/applications/editors/neovim/glrnvim/default.nix]: For GitHub sources you can use `fetchFromGitHub`.

File: pkgs/applications/editors/neovim/glrnvim/default.nix [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+  nativeBuildInputs = [ makeWrapper ];
+
+  postInstall = ''
+    wrapProgram $out/bin/glrnvim \
+      --prefix PATH : ${stdenv.lib.makeBinPath [alacritty neovim]}
+  '';
+
+  meta = with stdenv.lib; {
+    description = "A GPU-accelerated neovim GUI";
+    homepage = https://github.com/beeender/glrnvim;
```
Comment (JohnAZoidberg) [inline pkgs/applications/editors/neovim/glrnvim/default.nix]:
```suggestion
    homepage = "https://github.com/beeender/glrnvim";
```
We're probably going to deprecate unquoted URL syntax once RFC 45 is approved.

**Review by JohnAZoidberg — COMMENTED** [review]
Not sure this package is all that useful. It just runs neovim inside the terminal (alacritty, which does the GPU acceleration).
See: beeender/glrnvim#3
I'm not sure whether we have a policy for what packages to include, whether we want everything or not.

**Comment by sgraf812** [comment]
> Not sure this package is all that useful

Makes sense. I'll put in my NUR instead, then.

File: pkgs/top-level/all-packages.nix [hunk 3]
```diff
@@ -20574,6 +20574,8 @@ in
 
   neovim-qt = libsForQt5.callPackage ../applications/editors/neovim/qt.nix { };
 
+  glrnvim = libsForQt5.callPackage ../applications/editors/neovim/glrnvim { };
```
Comment (JohnAZoidberg) [inline pkgs/top-level/all-packages.nix]:
```
glrnvim = callPackage ../applications/editors/neovim/glrnvim { };
```

Also it's not a QT package, so you should just call the top-level `callPackage`.

**Pull request closed without merging**